superset/dashboards/commands/bulk_delete.py

```diff
@@ -40,7 +40,7 @@ def __init__(self, user: User, model_ids: List[int]):
         self._model_ids = model_ids
         self._models: Optional[List[Dashboard]] = None
 
-    def run(self):
+    def run(self) -> None:
```

Comment:
Should this be `Optional[Model]` per https://github.com/apache/incubator-superset/pull/9416.

Reply:
My take is that the abstract is defined has `Optional[Model]` and the implementations can return `None` or `Model`. If we set (for example) `create.run` -> Optional[Model]` then I should check for the model's existence on the return from: https://github.com/apache/incubator-superset/blob/master/superset/dashboards/api.py#L164. 

And I'm assuming that the commands succeed and return or raise. What do you think?

Reply:
On the other hand we can make them more coherent and always expect `Optional[Model]` this can make future implementation more safely "blind" to the command ifself, just run `run`

Reply:
I prefer the later, i.e., all the `run` signatures having the same return type. It helps to ensures consistency both now and for future classes.

Reply:
Ended up reverting typing on command to more explicit type, following @villebro comments and reasoning around:
- We want to migrate logic implemented on models to commands and the return types for `Command.run()` will vary a lot
- `Optional[Model]` will not cover all future use cases and will force us to implement logic that is unnecessary such has testing if the `run` method returned something, when it will always return of raise
- We will not be able to implement a single return type for all `Command.run()` unless creating a very big type definition

Reply:
@dpgaspar would you mind updating this [line](https://github.com/apache/incubator-superset/blob/master/superset/commands/base.py#L29) as well to ensure consistency?